machine_configuration/mco-update-boot-images.adoc

```diff
@@ -0,0 +1,61 @@
+:_content-type: ASSEMBLY
+:context: machine-configs-configure
+[id="mco-update-boot-images"]
+= Updating boot images
+include::_attributes/common-attributes.adoc[]
+
+toc::[]
+
+The Machine Config Operator (MCO) uses a boot image to start up a {op-system-first} node. By default, {product-title} does not manage the boot image.
+
+This means that the boot image in your cluster is not updated along with your cluster. For example, if your cluster was originally created with {product-title} 4.12, the boot image that the cluster uses to create nodes is the same 4.12 version, even if your cluster is at a later version. If the cluster is later upgraded to 4.13 or later, new nodes continue to scale with the same 4.12 image.
+
+This process could cause the following issues:
+
+* Extra time to start up nodes
+* Certificate expiration issues
```

Comment:
nit: certificate expiration shouldn't be affected by the bootimage version, should we remove this?

Reply:
@yuqi-zhang I think I will address this and your [other comment below](https://github.com/openshift/openshift-docs/pull/61580#discussion_r1651749564) in a separate issue.

Reply:
Works for me, please ping me on that when it's ready and I can help take a look, thanks!

Reply:
@yuqi-zhang The "certificate expiration issues"  came from @djoshy enhancement doc: 

[In certain long lived clusters, the MCS TLS cert contained within the above Ignition configuration may be out of date.](https://github.com/openshift/enhancements/pull/1496/files#diff-5299368e7b271ee5543c2c6a36b98e41445f0bc8200bc4521c10a3e44596bec1R50)

Does this make sense in that context?

Reply:
@yuqi-zhang @djoshy I might have missed this. When a new node is rolled out with an earlier boot image, does the image get updated to match the cluster? That seems to be the suggestion in some internal communications.  However, I didn't see this when I tested the feature. Maybe I didn't wait long enough?

Reply:
@yuqi-zhang might have a better idea about this - but I think that bit in the enhancement was about it helping with the reference to the TLS managed-secret. So while this boot image epic makes that a bit easier by always using the managed secret, it does not solve the TLS cert management problem by itself. 

> When a new node is rolled out with an earlier boot image, does the image get updated to match the cluster? 

A new node will always use the boot image in the machineset it was scaled from. By boot image here, we mean the image the node was born on, during its very first boot up. For an existing node, you cannot update the boot image once a node has been born, as it is already _alive_. You can only update the boot image of _future_ nodes that you scale up.

Reply:
Right, there is tangential relationship here. It's probably fine to talk about the certificate in the context of the bootimage, but there are other issues we need to solve as @djoshy points to, so it's not going to solve everything.